www/components/CreateRoot.tsx

```diff
@@ -35,23 +35,41 @@ export default function CreateRoot() {
     status,
     create,
   } = useCreateMerkleRoot()
+  const [hasDuplicatesErr, setHasDuplicates] = useState(false)
   const [addressInput, addressInputSet] = useState('')
 
+  useEffect(() => {
+    setHasDuplicates(false)
+  }, [addressInput])
+
   const handleSubmit = useCallback(() => {
     if (addressInput.trim().length === 0) {
       return
     }
 
-    const addresses = parseAddressesFromText(addressInput)
+    const { addresses, hasDuplicates } = parseAddressesFromText(
+      addressInput,
+      false,
+    )
+    if (hasDuplicates) {
+      setHasDuplicates(true)
+      return
+    }
+
     create(addresses)
   }, [addressInput, create])
 
+  const removeDuplicates = () => {
+    const { addresses } = parseAddressesFromText(addressInput, true)
+    addressInputSet(addresses.join('\n'))
+  }
+
   const parsedAddresses = useMemo(() => {
     if (addressInput.trim().length === 0) {
       return []
     }
 
-    return parseAddressesFromText(addressInput)
+    return parseAddressesFromText(addressInput, false).addresses
   }, [addressInput])
```

Comment:
can we change this to `const { addresses, hasDuplicates } = useMemo(...` and then drop the extra `hasDuplicates`  state?